helix-term/src/ui/explorer.rs

```diff
@@ -0,0 +1,1527 @@
+        let folder_path = self.nearest_folder()?;
+        self.prompt = Some((
+            PromptAction::CreateFileOrFolder,
+            Prompt::new(
+                format!(
+                    " New file or folder (ends with '{}'): ",
+                    std::path::MAIN_SEPARATOR
+                )
+                .into(),
+                None,
+                ui::completers::none,
+                |_, _, _| {},
+            )
+            .with_line(format!("{}/", folder_path.to_string_lossy()), cx.editor),
+        ));
+        Ok(())
+    }
+
+    fn nearest_folder(&self) -> Result<PathBuf> {
+        let current = self.tree.current()?.item();
+        if current.is_parent() {
+            Ok(current.path.to_path_buf())
+        } else {
+            let parent_path = current.path.parent().ok_or_else(|| {
+                anyhow::anyhow!(format!(
+                    "Unable to get parent path of '{}'",
+                    current.path.to_string_lossy()
+                ))
+            })?;
+            Ok(parent_path.to_path_buf())
+        }
+    }
+
+    fn new_remove_prompt(&mut self) -> Result<()> {
+        let item = self.tree.current()?.item();
+        match item.file_type {
+            FileType::Folder => self.new_remove_folder_prompt(),
+            FileType::File => self.new_remove_file_prompt(),
+            FileType::Root => bail!("Root is not removable"),
+        }
+    }
+
+    fn new_rename_prompt(&mut self, cx: &mut Context) -> Result<()> {
+        let path = self.tree.current_item()?.path.clone();
+        self.prompt = Some((
+            PromptAction::RenameFile,
+            Prompt::new(
+                " Rename to ".into(),
+                None,
+                ui::completers::none,
+                |_, _, _| {},
+            )
+            .with_line(path.to_string_lossy().to_string(), cx.editor),
+        ));
+        Ok(())
+    }
+
+    fn new_remove_file_prompt(&mut self) -> Result<()> {
+        let item = self.tree.current_item()?;
+        ensure!(
+            item.path.is_file(),
+            "The path '{}' is not a file",
+            item.path.to_string_lossy()
+        );
+        self.prompt = Some((
+            PromptAction::RemoveFile,
+            Prompt::new(
+                format!(" Delete file: '{}'? y/n: ", item.path.display()).into(),
+                None,
+                ui::completers::none,
+                |_, _, _| {},
+            ),
+        ));
+        Ok(())
+    }
+
+    fn new_remove_folder_prompt(&mut self) -> Result<()> {
+        let item = self.tree.current_item()?;
+        ensure!(
+            item.path.is_dir(),
+            "The path '{}' is not a folder",
+            item.path.to_string_lossy()
+        );
+
+        self.prompt = Some((
+            PromptAction::RemoveFolder,
+            Prompt::new(
+                format!(" Delete folder: '{}'? y/n: ", item.path.display()).into(),
```

Comment:
Shouldn't there be a default that is capitalized?

Reply:
Yup, the N should be capitalized.

Reply:
@pickfire We should have a `Select` component like HTML `<select>`; piggybacking the choice UX on `Prompt` will not scale well.